code/game/turfs/simulated/floor/lava_pool.dm

```diff
@@ -0,0 +1,45 @@
+/turf/simulated/floor/lava_pool
+		icon_state = "lava"
+		name = "lava pool"
+		desc = "A deadly pool of molten rock that incinerates almost anything on contact."
+		luminosity = 3
+
+/turf/simulated/floor/lava_pool/Entered(var/mob/living/carbon/C)
+	if(!istype(C))
+		return
+
+	SSobj.processing |= src
+	if(istype(C, /mob/living/carbon/human))
+		if(check_suit(C))
+			C.visible_message("<span class='warning'>Your reinforced suit protects you from [src]! </span>")
+			return
+
+	apply_lava_damage(C)
+	C.emote("scream")
```

Comment:
Just tested. It seems emote checks if they're dead so it's fine. 

The combination of text messages causes a bit of spam when dragging people through lava in a mining hardsuit so I'll look into making those less frequent.